hack/build-go.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# This script sets up a go workspace locally and builds all go components.
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
+
+function cleanup() {
+	return_code=$?
+	os::util::describe_return_code "${return_code}"
+	exit "${return_code}"
+}
+trap "cleanup" EXIT
+
+build_targets=("$@")
+if [[ -z "$@" ]]; then
+	build_targets=("${OS_CROSS_COMPILE_TARGETS[@]}")
```

Comment:
what cross compile targets?

Reply:
dockerregistry.  Should be testable under any platform, nothing special about it.  IMAGE_COMPILE is for things that only work in images.

Reply:
then dockerregistry should be an IMAGE_COMPILE target imho.  We aren't going to fix it if it's broken on another platform, are we?  nor should pulling in a linux-only godep prevent us from building it.